pkg/daemon/update.go

```diff
@@ -93,9 +99,57 @@ func getNodeRef(node *corev1.Node) *corev1.ObjectReference {
 	}
 }
 
-// finalizeAndReboot is the last step in an update(), and it can also
-// be called as a special case for the "bootstrap pivot".
-func (dn *Daemon) finalizeAndReboot(newConfig *mcfgv1.MachineConfig) (retErr error) {
+func reloadService(name string) error {
+	_, err := runGetOut("systemctl", "reload", name)
+	return err
+}
+
+// performPostConfigChangeAction takes action based on what postConfigChangeAction has been asked.
+// For non-reboot action, it applies configuration, updates node's config and state.
+// In the end uncordon node to schedule workload.
+// If at any point an error occurs, we reboot the node so that node has correct configuration.
+func (dn *Daemon) performPostConfigChangeAction(postConfigChangeActions []string, configName string) error {
```

Comment:
wondering if there should be an event here?

Reply:
Hmm, as in, add events for non-reboot scenarios? (We do have reboot events I believe)

Reply:
yes exactly wondering if they are covered sufficiently  (sorry for not being clear!)

Reply:
(i can take or leave this was just wondering)

Reply:
Added some in the specific scenarios below:
`dn.recorder.Eventf(getNodeRef(dn.node), corev1.EventTypeNormal, "Reboot", "Config changes do not require reboot.")`
and
`dn.recorder.Eventf(getNodeRef(dn.node), corev1.EventTypeNormal, "Reboot", "Config changes do not require reboot. Service %s was reloaded.", serviceName)`

Should I also add some warnings for the failure cases? Or what would be a good way to handle those (they would get a reboot event regardless)

Reply:
we have events for failsdrain and fails to reconcile for ex:
`dn.logSystem("Reloading %s configuration failed, node will reboot instead. Error: %v", serviceName, err)`
maybe a similar one near?: `dn.logSystem("Reloading %s configuration failed, node will reboot instead. Error: %v", serviceName, err)`

Reply:
Did you mean this line: `dn.recorder.Eventf(getNodeRef(dn.node), corev1.EventTypeWarning, "FailedToDrain", failMsg)` ?
So basically for what I added, instead of `Normal` I added it as a warning instead?

Reply:
oh duh yes sorry for the bad copypasta, yes that is the example, and yeah i thought maaybe put a warning near the reloading failed logging you already have to show it failed